### ISSUE TITLE
npm ERR! Cannot read property '_updated' of undefined

### ISSUE THREAD

**Issue by qkboo** [npm/npm#12439]
I just search by npm what express version are available
But npm error "npm ERR! Cannot read property '_updated' of undefined"

below contents are *npm-debug.log*
--
```
0 info it worked if it ends with ok
1 verbose cli [ '/home/qkboo/.nvm/versions/node/v5.10.1/bin/node',
1 verbose cli   '/home/qkboo/.nvm/versions/node/v5.10.1/bin/npm',
1 verbose cli   'search',
1 verbose cli   'express' ]
2 info using npm@3.8.3
3 info using node@v5.10.1
4 silly mapToRegistry name -/all
5 silly mapToRegistry using default registry
6 silly mapToRegistry registry https://registry.npmjs.org/
7 silly mapToRegistry data Result {
7 silly mapToRegistry   raw: '-/all',
7 silly mapToRegistry   scope: null,
7 silly mapToRegistry   name: null,
7 silly mapToRegistry   rawSpec: '-/all',
7 silly mapToRegistry   spec: '/home/qkboo/express_v4/uploader_multer/-/all',
7 silly mapToRegistry   type: 'local' }
8 silly mapToRegistry uri https://registry.npmjs.org/-/all
9 info updateIndex /home/qkboo/.npm/registry.npmjs.org/-/all/.cache.json
10 verbose correctMkdir /home/qkboo/.npm correctMkdir not in flight; initializing
11 verbose stack TypeError: Cannot read property '_updated' of undefined
11 verbose stack     at /home/qkboo/.nvm/versions/node/v5.10.1/lib/node_modules/npm/lib/cache/update-ind
ex.js:61:26
11 verbose stack     at /home/qkboo/.nvm/versions/node/v5.10.1/lib/node_modules/npm/node_modules/gracefu
l-fs/polyfills.js:210:7
11 verbose stack     at FSReqWrap.oncomplete (fs.js:82:15)
12 verbose cwd /home/qkboo/express_v4/uploader_multer
13 error Linux 4.4.4-std-3
14 error argv "/home/qkboo/.nvm/versions/node/v5.10.1/bin/node" "/home/qkboo/.nvm/versions/node/v5.10.1/
bin/npm" "search" "express"
15 error node v5.10.1
16 error npm  v3.8.3
17 error Cannot read property '_updated' of undefined
18 error If you need help, you may report this error at:
18 error     <https://github.com/npm/npm/issues>
19 verbose exit [ 1, true ]
```

**Comment by AdamMcCormick** [comment]
I'm seeing the same. Upgraded to node v6 recently and this is the first thing I've tried to search for. Received this:
<pre>
$ npm search spin
npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/Cellar/node/6.0.0/bin/node" "/usr/local/bin/npm" "search" "spin"
npm ERR! node v6.0.0
npm ERR! npm  v3.8.8

npm ERR! Cannot read property '_updated' of undefined
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
</pre>

And the npm-debug.log
<pre>0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/Cellar/node/6.0.0/bin/node',
1 verbose cli   '/usr/local/bin/npm',
1 verbose cli   'search',
1 verbose cli   'spin' ]
2 info using npm@3.8.8
3 info using node@v6.0.0
4 silly mapToRegistry name -/all
5 silly mapToRegistry using default registry
6 silly mapToRegistry registry https://registry.npmjs.org/
7 silly mapToRegistry data Result {
7 silly mapToRegistry   raw: '-/all',
7 silly mapToRegistry   scope: null,
7 silly mapToRegistry   name: null,
7 silly mapToRegistry   rawSpec: '-/all',
7 silly mapToRegistry   spec: '-/all',
7 silly mapToRegistry   type: 'local' }
8 silly mapToRegistry uri https://registry.npmjs.org/-/all
9 info updateIndex /Users/awoodsmccormick/.npm/registry.npmjs.org/-/all/.cache.json
10 verbose correctMkdir /Users/awoodsmccormick/.npm correctMkdir not in flight; initializing
11 verbose stack TypeError: Cannot read property '_updated' of undefined
11 verbose stack     at /usr/local/lib/node_modules/npm/lib/cache/update-index.js:61:26
11 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/graceful-fs/polyfills.js:210:7
11 verbose stack     at FSReqWrap.oncomplete (fs.js:117:15)
12 verbose cwd /Users/awoodsmccormick/Development/AdobeAddon
13 error Darwin 14.5.0
14 error argv "/usr/local/Cellar/node/6.0.0/bin/node" "/usr/local/bin/npm" "search" "spin"
15 error node v6.0.0
16 error npm  v3.8.8
17 error Cannot read property '_updated' of undefined
18 error If you need help, you may report this error at:
18 error     <https://github.com/npm/npm/issues>
19 verbose exit [ 1, true ]</pre>

**Comment by ian-bartholomew** [comment]
+1

**Comment by jallen0927** [comment]
Same error on Ubuntu 16.04:
```
$ npm search gulp-sass
npm ERR! Linux 4.4.0-22-generic
npm ERR! argv "/home/shawn/.nvm/versions/node/v6.2.0/bin/node" "/home/shawn/.nvm/versions/node/v6.2.0/bin/npm" "search" "gulp-sass"
npm ERR! node v6.2.0
npm ERR! npm  v3.8.9

npm ERR! Cannot read property '_updated' of undefined
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
npm WARN Building the local index for the first time, please be patient
```
In npm-debug.log
```
0 info it worked if it ends with ok
1 verbose cli [ '/home/shawn/.nvm/versions/node/v6.2.0/bin/node',
1 verbose cli   '/home/shawn/.nvm/versions/node/v6.2.0/bin/npm',
1 verbose cli   'search',
1 verbose cli   'gulp-sass' ]
2 info using npm@3.8.9
3 info using node@v6.2.0
4 silly mapToRegistry name -/all
5 silly mapToRegistry using default registry
6 silly mapToRegistry registry https://registry.npmjs.org/
7 silly mapToRegistry data Result {
7 silly mapToRegistry   raw: '-/all',
7 silly mapToRegistry   scope: null,
7 silly mapToRegistry   name: null,
7 silly mapToRegistry   rawSpec: '-/all',
7 silly mapToRegistry   spec: '-/all',
7 silly mapToRegistry   type: 'local' }
8 silly mapToRegistry uri https://registry.npmjs.org/-/all
9 info updateIndex /home/shawn/.npm/registry.npmjs.org/-/all/.cache.json
10 verbose correctMkdir /home/shawn/.npm correctMkdir not in flight; initializing
11 verbose stack TypeError: Cannot read property '_updated' of undefined
11 verbose stack     at /home/shawn/.nvm/versions/node/v6.2.0/lib/node_modules/npm/lib/cache/update-index.js:61:26
11 verbose stack     at /home/shawn/.nvm/versions/node/v6.2.0/lib/node_modules/npm/node_modules/graceful-fs/polyfills.js:210:7
11 verbose stack     at FSReqWrap.oncomplete (fs.js:117:15)
12 verbose cwd /var/www
13 error Linux 4.4.0-22-generic
14 error argv "/home/shawn/.nvm/versions/node/v6.2.0/bin/node" "/home/shawn/.nvm/versions/node/v6.2.0/bin/npm" "search" "gulp-sass"
15 error node v6.2.0
16 error npm  v3.8.9
17 error Cannot read property '_updated' of undefined
18 error If you need help, you may report this error at:
18 error     <https://github.com/npm/npm/issues>
19 verbose exit [ 1, true ]
```

**Comment by ivanchuk** [comment]
For me was solved with `npm cache clean`

**Comment by andreportela** [comment]
I'm not sure, but maybe this is related to #12619

**Comment by reach4thelasers** [comment]
I've not been able to do an npm search for over a week because of this.  Any updates or solutions?

**Comment by andreportela** [comment]
@reach4thelasers have you tried to run [this](https://github.com/npm/npm/issues/12619#issuecomment-224547637)?

**Comment by reach4thelasers** [comment]
I'm using Windows not a UNIX-based system - I've not had a chance to look up the paths etc.  And its easier to just do searches and npmsearch.com rather than type all that every time :)

**Comment by Zearin** [comment]
👍 Happening to me too, on ~~OS X~~ <ins>macOS</ins> (doh!)

I tried `npm cache clean` followed by another search, and got something different:

```
npm search "gulp-"
npm WARN Building the local index for the first time, please be patient
⸨░░░░░░░░░░░░░░░░░░⸩ ⠏ : WARN Building the local index for the first time, please be patient
<--- Last few GCs --->

  121588 ms: Mark-sweep 860.3 (1122.4) -> 833.6 (1102.1) MB, 1340.2 / 0 ms [allocation failure] [GC in old space requested].
  123102 ms: Mark-sweep 833.6 (1102.1) -> 833.4 (1111.1) MB, 1514.2 / 0 ms [allocation failure] [GC in old space requested].
  124700 ms: Mark-sweep 833.4 (1111.1) -> 833.4 (1105.1) MB, 1598.1 / 0 ms [last resort gc].
  126509 ms: Mark-sweep 833.4 (1105.1) -> 833.3 (1097.1) MB, 1809.0 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x3310e7cc9e59 <JS Object>
    1: fromString(aka fromString) [buffer.js:234] [pc=0x22d74cd9b1a9] (this=0x3310e7c04189 <undefined>,string=0x3778b34bc3f1 <Very long string[178438742]>,encoding=0x3310e7cd79c9 <String[4]: utf8>)
    2: from [buffer.js:131] [pc=0x22d74cd3c396] (this=0x7c8d21e61f1 <JS Function Buffer (SharedFunctionInfo 0xe57a28f6641)>,value=0x3778b34bc3f1 <Very long string[178438742]>,encodingOrOffset=0x3310e...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
Abort trap: 6
```

**Comment by crashspringfield** [comment]
tried this on Mint 17.4 and got basically the same as @Zearin 


    <--- Last few GCs --->

       37734 ms: Mark-sweep 861.4 (1127.4) -> 835.9 (1113.1) MB, 666.9 / 0 ms [allocation failure] [GC in old space requested].
       38742 ms: Mark-sweep 835.9 (1113.1) -> 835.8 (1115.1) MB, 1007.8 / 0 ms [allocation failure] [GC in old space requested].
       39479 ms: Mark-sweep 835.8 (1115.1) -> 835.8 (1107.1) MB, 736.9 / 0 ms [last resort gc].
       40309 ms: Mark-sweep 835.8 (1107.1) -> 835.8 (1099.1) MB, 829.7 / 0 ms [last resort gc].


    <--- JS stacktrace --->

    ==== JS stack trace =========================================

    Security context: 0x2f56619c9e59 <JS Object>
        1: fromString(aka fromString) [buffer.js:234] [pc=0x2820340eafe9] (this=0x2f5661904189    <undefined>,string=0xe4ebc24d19 <Very long string[179116197]>,encoding=0x2f56619d7aa9  <String[4]: utf8>)
        2: from [buffer.js:131] [pc=0x2820340672f6] (this=0x12c4091dd6e9 <JS Function Buffer (SharedFunctionInfo 0x34a5e40f2651)>,value=0xe4ebc24d19 <Very long string[179116197]>,encodingOrOffset=0x2f56619...

    FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
    Aborted

**Comment by jallen0927** [comment]
For me, the error "npm ERR! Cannot read property '_updated' of undefined" is resolved by doing "npm clear cache". And the memory error got solved by what @andreportela suggested [this](https://github.com/npm/npm/issues/12619#issuecomment-224547637)

**Comment by pcgeek86** [comment]
Got the same thing on my Windows 10 system.

```
PS C:\Users\TrevorSullivan> npm -v
3.10.3
PS C:\Users\TrevorSullivan> (Get-CimInstance -ClassName Win32_OperatingSystem).Version
10.0.10586
```

Here's the log:

```
npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "search" "aws"
npm ERR! node v6.3.0
npm ERR! npm  v3.10.3

npm ERR! Cannot read property '_updated' of undefined
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
npm WARN Building the local index for the first time, please be patient

<--- Last few GCs --->

   28639 ms: Mark-sweep 876.0 (1146.5) -> 875.3 (1152.5) MB, 580.5 / 0 ms (+ 64.6 ms in 87 steps since start of marking, biggest step 11.6 ms) [allocation fai
lure] [GC in old space requested].
   29469 ms: Mark-sweep 875.3 (1152.5) -> 850.2 (1130.1) MB, 830.2 / 0 ms [allocation failure] [GC in old space requested].
   30385 ms: Mark-sweep 850.2 (1130.1) -> 850.2 (1119.1) MB, 916.3 / 0 ms [last resort gc].
   31281 ms: Mark-sweep 850.2 (1119.1) -> 850.2 (1112.1) MB, 895.3 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 00000152EC5C9E79 <JS Object>
    1: fromString(aka fromString) [buffer.js:194] [pc=000001E5487E99F0] (this=00000152EC504189 <undefined>,string=0000035A07B4D0D9 <Very long string[182398036
]>,encoding=00000152EC5D6381 <String[4]: utf8>)
    2: from [buffer.js:94] [pc=000001E5487CC636] (this=000001B7DD5A7149 <JS Function Buffer (SharedFunctionInfo 00000152EC5E3541)>,value=0000035A07B4D0D9 <Ver
y long string[182398036]>,encod...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
```

Also:

```
npm WARN Building the local index for the first time, please be patient
[..................] - : WARN Building the local index for the first time, please be patient
<--- Last few GCs --->

   28444 ms: Mark-sweep 877.1 (1264.5) -> 875.6 (1272.5) MB, 395.5 / 0 ms (+ 303.5 ms in 1590 steps since start of marking, biggest step 8.7 ms) [allocation f
ailure] [GC in old space requested].
   29233 ms: Mark-sweep 875.6 (1272.5) -> 850.4 (1252.1) MB, 789.2 / 0 ms [allocation failure] [GC in old space requested].
   30027 ms: Mark-sweep 850.4 (1252.1) -> 850.4 (1246.1) MB, 793.6 / 0 ms [last resort gc].
   30805 ms: Mark-sweep 850.4 (1246.1) -> 850.3 (1239.1) MB, 778.3 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 000003303CBC9E79 <JS Object>
    1: fromString(aka fromString) [buffer.js:194] [pc=0000016DED6DF550] (this=000003303CB04189 <undefined>,string=000002D666CE4649 <Very long string[182398036
]>,encoding=000003303CBD6381 <String[4]: utf8>)
    2: from [buffer.js:94] [pc=0000016DED633636] (this=0000037A854AA0F1 <JS Function Buffer (SharedFunctionInfo 000003303CBE3541)>,value=000002D666CE4649 <Ver
y long string[182398036]>,encod...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
```

**Comment by lyonsun** [comment]
+1

**Comment by silvenon** [comment]
`npm cache clean` solved it for me, too.

**Comment by spinningarrow** [comment]
Same issue, OS X El Capitan 10.11.6, Node v6.3.0, npm v3.10.6.
`npm cache clean` did not help.

**Comment by faddat** [comment]
```
npm search hyperterm
npm WARN Building the local index for the first time, please be patient
[..................] \ : WARN Building the local index for the first time, please be patient
<--- Last few GCs --->

   25246 ms: Mark-sweep 888.2 (1165.6) -> 887.6 (1169.6) MB, 337.1 / 0 ms (+ 83.5 ms in 400 steps since start of marking, biggest step 5.7 ms) [allocation failure] [GC in old space requested].
   25785 ms: Mark-sweep 887.6 (1169.6) -> 862.4 (1145.1) MB, 539.4 / 0 ms [allocation failure] [GC in old space requested].
   26334 ms: Mark-sweep 862.4 (1145.1) -> 862.4 (1137.1) MB, 549.0 / 0 ms [last resort gc].
   26898 ms: Mark-sweep 862.4 (1137.1) -> 862.3 (1131.1) MB, 563.7 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x77608bc9e31 <JS Object>
    1: fromString(aka fromString) [buffer.js:194] [pc=0x1e95a3deacd0] (this=0x77608b04189 <undefined>,string=0x354236a64c99 <Very long string[187956334]>,encoding=0x77608bd7a21 <String[4]: utf8>)
    2: from [buffer.js:94] [pc=0x1e95a3d6c2b6] (this=0x20e6b22cd811 <JS Function Buffer (SharedFunctionInfo 0x77608be32f9)>,value=0x354236a64c99 <Very long string[187956334]>,encodingOrOffset=0x77608bd7...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [npm]
 2: 0xfb1c8c [npm]
 3: v8::Utils::ReportApiFailure(char const*, char const*) [npm]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [npm]
 5: v8::internal::Factory::NewRawTwoByteString(int, v8::internal::PretenureFlag) [npm]
 6: v8::internal::String::SlowFlatten(v8::internal::Handle<v8::internal::ConsString>, v8::internal::PretenureFlag) [npm]
 7: v8::internal::String::Flatten(v8::internal::Handle<v8::internal::String>, v8::internal::PretenureFlag) [npm]
 8: v8::String::WriteUtf8(char*, int, int*, int) const [npm]
 9: node::StringBytes::Write(v8::Isolate*, char*, unsigned long, v8::Local<v8::Value>, node::encoding, int*) [npm]
10: node::Buffer::New(v8::Isolate*, v8::Local<v8::String>, node::encoding) [npm]
11: node::Buffer::CreateFromString(v8::FunctionCallbackInfo<v8::Value> const&) [npm]
12: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&)) [npm]
13: 0x9d5b0b [npm]
14: 0x9d60b1 [npm]
15: 0x1e95a390961b
Aborted (core dumped)
```

It's real.  npm clean cache is not giving it any love.....

**Comment by rosen-vladimirov** [comment]
Ok, I've hit the same issue on my side and it looks like it is related to Node.js (v8) memory.
Steps to reproduce:
* `$ npm cache clean` - make sure the .cache.json file in `<npm cache dir>/registry.npmj.org/-/all/.cache.json` is deleted.
* `$ npm search lodash` - this will create the .cache.json file and will fail with:
```
$ npm search lodash
npm WARN Building the local index for the first time, please be patient

<--- Last few GCs --->

   41419 ms: Mark-sweep 922.6 (1223.8) -> 921.9 (1231.8) MB, 507.9 / 0 ms (+ 184.3 ms in 1095 steps since start of marking, biggest step 8.5 ms) [allocation failure] [GC in old space requested].
   42282 ms: Mark-sweep 921.9 (1231.8) -> 896.6 (1210.1) MB, 863.3 / 0 ms [allocation failure] [GC in old space requested].
   43164 ms: Mark-sweep 896.6 (1210.1) -> 896.6 (1204.1) MB, 881.4 / 0 ms [last resort gc].
   44021 ms: Mark-sweep 896.6 (1204.1) -> 896.5 (1198.1) MB, 857.0 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0000036FA76C9E51 <JS Object>
    1: fromString(aka fromString) [buffer.js:194] [pc=00000082981398B0] (this=0000036FA7604189 <undefined>,string=000002830F8A53C1 <Very long string[193156529]>,encoding=0000036FA76D6379 <String[4]: utf8>)
    2: from [buffer.js:94] [pc=00000082981BA236] (this=0000013D4A2A3F29 <JS Function Buffer (SharedFunctionInfo 0000036FA76E2839)>,value=000002830F8A53C1 <Very long string[193156529]>,encod...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
```
* check the file .cache.json now - it has empty object in it ` {} `. Now run `$ npm search lodash` again. The error for `Cannot read property '_updated' of undefined` is raised again and new search is started.
```
$ npm search lodash
npm ERR! Windows_NT 6.3.9600
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "search" "lodash"
npm ERR! node v6.4.0
npm ERR! npm  v3.10.3

npm ERR! Cannot read property '_updated' of undefined
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
npm WARN Building the local index for the first time, please be patient

<--- Last few GCs --->

   38391 ms: Mark-sweep 922.6 (1227.8) -> 922.0 (1235.8) MB, 489.8 / 0 ms (+ 180.7 ms in 1085 steps since start of marking, biggest step 9.6 ms) [allocation failure] [GC in old space requested].
   39261 ms: Mark-sweep 922.0 (1235.8) -> 896.7 (1213.1) MB, 870.0 / 0 ms [allocation failure] [GC in old space requested].
   40123 ms: Mark-sweep 896.7 (1213.1) -> 896.7 (1207.1) MB, 862.2 / 0 ms [last resort gc].
   40973 ms: Mark-sweep 896.7 (1207.1) -> 896.7 (1200.1) MB, 850.4 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 000000DFCFDC9E51 <JS Object>
    1: fromString(aka fromString) [buffer.js:194] [pc=00000165612E7070] (this=000000DFCFD04189 <undefined>,string=000002666A154EC1 <Very long string[193156529]>,encoding=000000DFCFDD6379 <String[4]: utf8>)
    2: from [buffer.js:94] [pc=00000165612CCF96] (this=0000003FE18A39D9 <JS Function Buffer (SharedFunctionInfo 000000DFCFDE2839)>,value=000002666A154EC1 <Very long string[193156529]>,encod...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
```

I've created a very simple repro using the `request` module (which is used by npm internally):
```JavaScript
var request = require('request');
request('http://registry.npmjs.org/-/all', function (error, response, body) {
  console.log("before");
  console.log(response);
  console.log("after");
});
```

If I try to use the error or body object from the callback, everything works as expected. But trying to use the response object leads to the error.

**Comment by rosen-vladimirov** [comment]
ping @othiym23 @iarna - do you think the steps above are enough, so the `needs-repro` tag can be removed?

**Comment by zkat** [comment]
This will probably be addressed by https://github.com/npm/npm/pull/13746

**Comment by rosen-vladimirov** [comment]
@zkat Thank you, most probably this will help. But we've investigated the issue further and it definitely is a limitation of v8 memory usage. I've tried my sample script with request module, by passing `--max-old-space-size=2048` argument to Node.js and it started working. So maybe npm binary should pass such argument to Node.js?

**Comment by rosen-vladimirov** [comment]
Also I forgot to mention my workaround - I've manually edited the `npm.cmd` executable and added the --max-old-space-size=2048, so my last line of the `npm.cmd` is now:
```
"%NODE_EXE%" --max-old-space-size=2048 "%NPM_CLI_JS%" %*
```
This way I was able to find the plugins.

**Comment by pavelz** [comment]
vim `which npm` 

change:
`#!/usr/bin/env node
to:
`#!/usr/bin/env node --max-old-space-size=2048

**Comment by muescha** [comment]
i had the same problem


```bash
$ npm search hpm
npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/Cellar/node/6.7.0/bin/node" "/usr/local/bin/npm" "search" "hpm"
npm ERR! node v6.7.0
npm ERR! npm  v3.10.8

npm ERR! Cannot read property '_updated' of undefined
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
npm WARN Building the local index for the first time, please be patient

<--- Last few GCs --->

  246367 ms: Mark-sweep 1016.6 (1310.2) -> 990.7 (1292.1) MB, 905.0 / 0.0 ms [allocation failure] [GC in old space requested].
  247391 ms: Mark-sweep 990.7 (1292.1) -> 990.7 (1293.1) MB, 1024.2 / 0.0 ms [allocation failure] [GC in old space requested].
  248541 ms: Mark-sweep 990.7 (1293.1) -> 990.6 (1256.1) MB, 1148.0 / 0.0 ms [last resort gc].
  249659 ms: Mark-sweep 990.6 (1256.1) -> 990.6 (1249.1) MB, 1118.0 / 0.0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x38d7032cfb51 <JS Object>
    1: fromString(aka fromString) [buffer.js:195] [pc=0x112c616e8750] (this=0x38d703204381 <undefined>,string=0x3ebd3720c2e1 <Very long string[206333869]>,encoding=0x38d7032dd329 <String[4]: utf8>)
    2: from [buffer.js:95] [pc=0x112c616951a2] (this=0x893322a8ef1 <JS Function Buffer (SharedFunctionInfo 0x3f2b3d7f6389)>,value=0x3ebd3720c2e1 <Very long string[206333869]>,encodingOrOffset=0x38d70...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [/usr/local/bin/node]
 2: node::FatalException(v8::Isolate*, v8::Local<v8::Value>, v8::Local<v8::Message>) [/usr/local/bin/node]
 3: v8::Utils::ReportApiFailure(char const*, char const*) [/usr/local/bin/node]
 4: v8::Utils::ApiCheck(bool, char const*, char const*) [/usr/local/bin/node]
 5: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [/usr/local/bin/node]
 6: v8::internal::Factory::NewRawTwoByteString(int, v8::internal::PretenureFlag) [/usr/local/bin/node]
 7: v8::internal::String::SlowFlatten(v8::internal::Handle<v8::internal::ConsString>, v8::internal::PretenureFlag) [/usr/local/bin/node]
 8: v8::String::WriteUtf8(char*, int, int*, int) const [/usr/local/bin/node]
 9: node::StringBytes::Write(v8::Isolate*, char*, unsigned long, v8::Local<v8::Value>, node::encoding, int*) [/usr/local/bin/node]
10: node::Buffer::New(v8::Isolate*, v8::Local<v8::String>, node::encoding) [/usr/local/bin/node]
11: node::Buffer::CreateFromString(v8::FunctionCallbackInfo<v8::Value> const&) [/usr/local/bin/node]
12: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&)) [/usr/local/bin/node]
13: v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::(anonymous namespace)::BuiltinArguments<(v8::internal::BuiltinExtraArguments)1>) [/usr/local/bin/node]
14: v8::internal::Builtin_Impl_HandleApiCall(v8::internal::(anonymous namespace)::BuiltinArguments<(v8::internal::BuiltinExtraArguments)1>, v8::internal::Isolate*) [/usr/local/bin/node]
15: v8::internal::Builtin_HandleApiCall(int, v8::internal::Object**, v8::internal::Isolate*) [/usr/local/bin/node]
16: 0x112c612092a7
Abort trap: 6
$
```

@pavelz Solution works for me

**Comment by jankun** [comment]
@pavelz: 

it should be `--max_old_space_size=2048` **instead of**  --max-old-space-size=2048 ?

$ node --v8-options | grep max
**--max_old_space_size** (max size of the old space (in Mbytes))

**Comment by glangston83** [comment]
Same error: 
➜  kestrel4 git:(eng-piechart-component) ✗ npm search aws
npm ERR! Darwin 16.0.0
npm ERR! argv "/usr/local/Cellar/node/6.7.0/bin/node" "/usr/local/bin/npm" "search" "aws"
npm ERR! node v6.7.0
npm ERR! npm  v3.10.8

npm ERR! Cannot read property '_updated' of undefined

@pavelz Solution works for me

**Comment by ownmaster** [comment]
None of the above solutions worked for me:
1) `npm cache clean`
2) `npm search jquery --max_old_space_size=2048`

Log:
```
17 http request GET https://registry.npmjs.org/-/all
18 http 200 https://registry.npmjs.org/-/all
...
19 verbose headers   'x-timer': 'S1481817637.522351,VS0,VE0' }
20 verbose stack TypeError: Cannot read property '_updated' of undefined
...
22 error Windows_NT 10.0.14393
```

**Comment by KenanY** [comment]
@ownmaster I believe this was resolved in `npm@4.0.0` :)
